src/handlers.cpp

```diff
@@ -1043,14 +1043,34 @@ ImpuRegDataTask::RequestType ImpuRegDataTask::request_type_from_body(std::string
   return ret;
 }
 
+std::string ImpuRegDataTask::server_name_from_body(std::string body)
+{
+  TRC_DEBUG("Determining server name from '%s'", body.c_str());
+
+  rapidjson::Document document;
+  document.Parse<0>(body.c_str());
+
+  if (!document.IsObject() ||
+      !document.HasMember("server_name") ||
+      !document["server_name"].IsString())
```

Comment:
This if statement contains a mixture of errors not errors - you could split these up to make debugging easier.